### PR TITLE
Catch process.TotalProcessorTime  exception

### PR DESCRIPTION
**Pull request by timmac-qmc**
process.TotalProcessorTime is not supported on all platforms
image

### CODE REVIEW

**Comment by konraddysput** [comment]
Thanks for reaching us to us with pull request! 

Calculating process.age can be done via internal calculation. The catch block supposes to catch other exceptions related to memory information. @perf2711 can you share your approach?

**Comment by timmac-qmc** [comment]
> The catch block supposes to catch other exceptions related to memory information. 

The code seems to contradict this statement considering you're explicitly catching "PlatformNotSupportedException"

File: Backtrace/Model/JsonData/BacktraceAttributes.cs
Comment (perf2711) [inline Backtrace/Model/JsonData/BacktraceAttributes.cs:244]:
```suggestion
            try
            {
                //How long the application has been running  in secounds
                var processAge = Math.Round(process.TotalProcessorTime.TotalSeconds);
                var totalProcessAge = unchecked((long)processAge);
                if (totalProcessAge > 0)
                {
                    Attributes["process.age"] = totalProcessAge;
                }
            }
            catch (NotSupportedException)
            {
                Trace.TraceWarning($"Cannot retrieve process age - TotalProcessorTime is not supported");
            }
            
            try
            {            
```

**Review by perf2711 — COMMENTED** [review]
Thank you for your submission!

This change is sound. The only thing I could suggest is that we keep try/catch separate, so we can still try and retrieve other stuff.

As for the alternative approach, we can implement this after this is merged, so as not to block this change.

**Comment by perf2711** [comment]
@timmac-qmc can you fix formatting issues, so we can merge this change? See [this check](https://github.com/backtrace-labs/backtrace-csharp/actions/runs/10788805198/job/29924008223?pr=44).

**Comment by timmac-qmc** [comment]
> @timmac-qmc can you fix formatting issues, so we can merge this change? See [this check](https://github.com/backtrace-labs/backtrace-csharp/actions/runs/10788805198/job/29924008223?pr=44).

Should hopefully be correct now

**Comment by timmac-qmc** [comment]
I can't see why it's failing, formatting looks fine. Spaces for indentation seems an odd choice...

**Review by perf2711 — APPROVED** [review]

**Pull request merged**
